practice-two/src/services/customerService.js

```diff
@@ -1,12 +1,58 @@
-// import { BASE_URL, PATH } from "@constants";
+import { BASE_URL, PATH } from '@constants';
 
+// Get customer list
 const getAllCustomerService = async (url) => {
   const response = await fetch(url);
   if (!response.ok) {
-    throw new Error("An error occurred while fetching the data.");
+    throw new Error('An error occurred while fetching the data.');
   }
   const data = await response.json();
   return data;
 };
 
-export default getAllCustomerService;
+const service = async (path, method, data) => {
+  const url = `${BASE_URL}/${path}`;
+  try {
+    const response = await fetch(url, {
+      method,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (response.ok) {
+      return { data: await response.json(), error: null };
```

Comment:
create variable cache response.json() to reuse

Reply:
![image](https://github.com/datdragon777/React-Training/assets/74806826/93110fa4-ea5c-4d53-9720-00e77eb7471f)